### PR TITLE
Create Issue and PR template files

### PR DESCRIPTION
**Pull request by ThomDietrich**
This is a first idea of providing a pull request template.
My main motivation for this PR is to motivate users to clarify the impact of their PR better.

The document is just a first draft! Please add your comments.

A few notes:
* The purpose of the html comment block is to make sure the template doesn't show up in the PR
* Analogue to the included file a file "ISSUE_TEMPLATE.md" can be defined.
* Both files plus the file CONTRIBUTING.md can alternatively be moved to a folder ".github". 

Signed-off-by: Thomas Dietrich <thomas.dietrich@tu-ilmenau.de> (github: ThomDietrich)

### CODE REVIEW

**Comment by martinvw** [comment]
Great idea!

Should we maybe also mention:
* the sign-off 
* what the github checks mean
* something about the static code analysis
* a reminder about coding / development guidelines

**Comment by ThomDietrich** [comment]
Hey Martin!
I tried to not copy what's already mentioned in Contributing.md or the Style Guide page. We could include links to the respective documents!?

**Comment by ThomDietrich** [comment]
@martinvw are you already a maintainer for this repo? Please feel free to go through the document once, add your ideas and wording corrections and commit them.

**Comment by martinvw** [comment]
Fair, good idea to put a reference to it.

**Comment by martinvw** [comment]
The reason I mentioned it as that the following happens quite often:

https://github.com/openhab/openhab2-addons/pull/2289#issuecomment-302922930

So a checklist when creating a PR does feel like a good idea, but if we have a clear checklist somewhere else and refer to that in a clear way I'm fine too.

I'll put it on my list :-)

**Comment by ThomDietrich** [comment]
Oh wow I didn't know that ^^ Would have made this way easier...

As for the squashing/signing/"fixes...": I think we can add them as a short note and still link to other sources for more just as you did in the example. Are you going to add this or should I?

**Comment by martinvw** [comment]
I'm in the office and working now (at least was 15 seconds ago ;-) ) so it would be great if you can pick it up 👍

**Comment by ThomDietrich** [comment]
same here... ;)

**Comment by martinvw** [comment]
I'll start with it now :-)

**Comment by martinvw** [comment]
@ThomDietrich please check it out :-)

@steve-bate I copied the openhab1-addons ISSUE_TEMPLATE.md I assume its license is compatible but if you have any problems with it, please let me know ASAP

**Comment by ThomDietrich** [comment]
@martinvw great additions! I went through both documents again. While a bit of guidance is a good idea in the Issues section, I've removed the sections from the PR template, I think it's better to allow the submitter to describe their change in their own words. Please speak freely if you disagree and we can look for a compromise. All other changes are formatting details and attempts to reduce the character count. 

I think we are ready.

@kaikreuzer any further comments?

File: PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+<!-- Examples: -->
+<!-- - "[homematic] Improve communication with weak signal devices" -->
+<!-- - "[timemachine][WIP] Initial contribution" -->
+<!-- - "Update contribution guidelines on new signing rules" -->
+
+<!-- DESCRIPTION -->
+
+<!-- Please give a few sentences describing the overall goals of the pull request. -->
+<!-- Give enough details to make the improvement and changes of the PR understandable  -->
+<!-- to both developers and tech-savy users. The description can contain but is not limited to: -->
```
Comment (martinvw) [inline PULL_REQUEST_TEMPLATE.md]: I just notice that the list below is (or became) inconsistent, the first three items have a different format and type as the others the later or more questions / check maybe we should rephrase the first three to the same style.

Also we should maybe change the sentence "The description can contain but is not limited to" to something like, please check whether your contribution against the following checklist of something similar.

**Comment by kaikreuzer** [comment]
> @kaikreuzer any further comments?

Not from me, looks great and will definitely help improving issue/PR quality.
So @martinvw, once you are happy, feel free to merge!

**Comment by martinvw** [comment]
@ThomDietrich I'm working on the rephasing now, almost finished...

**Comment by martinvw** [comment]
@ThomDietrich should we move it to the `.github` folder?

**Comment by ThomDietrich** [comment]
One last change, I've split up the "for the end user" part because I feel this should be emphasized in order to be able to inform users of changes that may influence or harm their existing setup.

Regarding the .github folder: I'll you have decide that, I think root is just fine.

Ready ;)

**Comment by martinvw** [comment]
Sometimes mention bot seems to take a break, it should still be active. I just checked the community thread.

What is the status about your 'conclusion' there:

> I have to take one step back. The last commit in the original Haunt repository was done in 2012. There are forks with commits in 2016 but I'd have to reevaluate the fitness of these. This doesn't change the fact, that tools like this exist and might be an option for openhab.

**Comment by ThomDietrich** [comment]
@martinvw Let's continue over in the community. https://community.openhab.org/t/bot-for-github-issues-and-pull-requests-management/18361/13

**Pull request merged**